### ISSUE TITLE
Word borders don't seem to be working with `split`

### ISSUE THREAD

**Issue by adambadawy** [rust-lang-nursery/regex#204]
I expect this code:

```rust
fn main() {
    let re = Regex::new(r"\b").unwrap();
    let v: Vec<&str> = re.split("Should this (work?)").collect();
    println!("{:?}", v);
}
```

To output this:

```
["", "Should", " ", "this", " (", "work", "?)"]
```

But it outputs this:

```
["", "Should", " this", " ", "(work", "?", ")"]
```

## Meta

```
rustc 1.8.0 (db2939409 2016-04-11)
binary: rustc
commit-hash: db2939409db26ab4904372c82492cd3488e4c44e
commit-date: 2016-04-11
host: x86_64-apple-darwin
release: 1.8.0
```

**Comment by BurntSushi** [comment]
This is indeed a bug. It turns out to be a bug in how `\b` is being matched, which in turn is related to the recent change that `\b` is now speculatively matched by the DFA, which is getting it wrong. Working on it now.

**Comment by BurntSushi** [comment]
Yup. I was totally mishandling how word boundaries were being computed for start states. A fix is in PR #211.
